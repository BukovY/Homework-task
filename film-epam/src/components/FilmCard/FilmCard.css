.card .title{
    font-size: 18px;
}
.card .genres{
    font-size: 14px;
}
.film_cover {
    max-height: 290px;
}
.film_cover img {
width: 100%;

}
.film_cover .play_film {
    visibility: hidden;
    width: 90px;
    position: relative;
    bottom: 270px;
    left: calc(50% - 45px);

}
.film_cover:hover .play_film {
    visibility: visible;
    width: 90px;
    bottom: 200px;
    left: calc(50% - 45px);
    cursor: pointer;
}

.rating {
    position: relative;
    top: 35px;
    left: -5px;
    width: 30px;
    text-align: center;
    border-radius: 5px;
}
.rating_up {
    background-color: green;
}
.rating_down {
    background-color: gray;
}