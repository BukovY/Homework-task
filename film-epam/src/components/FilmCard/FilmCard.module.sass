.card
  color: white
  a
    text-decoration: none
  .title
    font-size: 18px
  .genres
    font-size: 14px

.film_cover
  width: 100%
  height: auto
  cursor: pointer
  position: relative
  aspect-ratio: 0.68
  img + img
    width: 50%
    border-radius: 5px
  &:hover .block
    visibility: visible


.block
  position: absolute
  left: 25%
  top: 30%
  padding: 5px
  display: block
  visibility: hidden

.rating
  position: relative
  top: 35px
  left: -5px
  width: 30px
  text-align: center
  border-radius: 5px
  z-index: 1

.rating_up
  background-color: green

.rating_down
  background-color: gray

.rating_hide
  color: #00ff0000
  background-color: #00ff0000

