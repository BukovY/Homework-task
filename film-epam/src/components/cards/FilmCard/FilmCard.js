import React from "react";

function FilmCard({props}) {
    return (<div>
            <p>{props.title}</p>
        </div>
    );
}

export default FilmCard;