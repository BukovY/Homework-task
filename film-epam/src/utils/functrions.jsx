export const genresIndexToString = (arr, map) => {
  const genresToRender = [];
  for (let i of map) {
    if (arr.indexOf(i.id) !== -1) {
      genresToRender.push(i.name);
    }
  }
  return genresToRender;
};

export const minToTime = (num) => {
  let min = num % 60;
  let hour = (num - min) / 60;
  return String(hour) + ":" + String(min);
};

export const getPhotoCard = (path) => {
  return path === null || path === undefined
    ? "https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-4-user-grey-d8fe957375e70239d6abdd549fd7568c89281b2179b5f4470e2e12895792dfa5.svg"
    : `https://image.tmdb.org/t/p/w1280/${path}`;
};

export const getPeopleCard = (img) => {
  return img === null || img === undefined
    ? "https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-4-user-grey-d8fe957375e70239d6abdd549fd7568c89281b2179b5f4470e2e12895792dfa5.svg"
    : `https://image.tmdb.org/t/p/w500/${img}`;
};

export const getFilmCover = (img) => {
  return img === null || img === undefined
    ? "https://upload.wikimedia.org/wikipedia/commons/6/65/No-Image-Placeholder.svg"
    : `https://image.tmdb.org/t/p/w500/${img}`;
};

export const getPaginationRange = (maxPage) =>
  Array(maxPage)
    .fill(1)
    .reduce((prev, next, index) => {
      return [...prev, index + 1];
    }, []);

export const getIndexLanguage = (lang) => {
  return lang === "EN" ? 0 : lang === "RU" ? 1 : 2;
};

export const matchOnlyNumber = (str) => /^\d+$/.test(str);
