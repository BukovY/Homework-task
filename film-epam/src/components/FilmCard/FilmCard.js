import React from "react";
import './FilmCard.css'
import play from '../../src/img/play.svg'

function FilmCard({img, rating, title, genres}) {

    return (
        <div className='card'>
            <div className={rating < 7 ? 'rating rating_down' : 'rating rating_up'}>{rating}</div>
            <div className='film_cover'>
                <img src={`https://image.tmdb.org/t/p/w500/${img}`} className='film_cover' alt={title} />
                <img src={play} width='50px' className='play_film' alt='play' />
            </div>
            {title? <div className='title'>{title}</div> : ''}
            {genres? <div className='genres'>{genres.join(" ")}</div> : ''}
        </div>
    );
}

export default FilmCard;