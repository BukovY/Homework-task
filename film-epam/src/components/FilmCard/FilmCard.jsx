import React from "react";
import "./FilmCard.css";
import play from "../../static/img/play.svg";
import { store } from "../../redux/store";
import { rerender } from "../../index";

const FilmCard = ({ img, rating, title, genres, id }) => {
  const openFilm = (id) => {
    store.selectedFilm = id;
    store.isTooltipLanguageOpen = false;
    store.selectPage = "movie";
    rerender();
  };
  if(img == null){
      img = 'https://upload.wikimedia.org/wikipedia/commons/6/65/No-Image-Placeholder.svg'
  } else {
      img = `https://image.tmdb.org/t/p/w500/${img}`
  }
  return (
    <div className="card">
      <div className={rating < 7 ? "rating rating_down" : "rating rating_up"}>
        {rating}
      </div>
      <div className="film_cover">
        <img
          src={img}
          className="film_cover"
          alt={title}
          onClick={() => openFilm(id)}
        />
        <img src={play} width="50px" className="play_film" alt="play" />
      </div>
      {title ? <div className="title">{title}</div> : ""}
      {genres ? <div className="genres">{genres.join(" ")}</div> : ""}
    </div>
  );
};

export default FilmCard;
