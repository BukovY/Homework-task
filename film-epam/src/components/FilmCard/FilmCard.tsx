import React from "react";
import play from "../../static/img/play.svg";
import styles from "./FilmCard.module.sass";
import { genresIndexToString, getFilmCover } from "../../utils/functrions";
import { useDispatch, useSelector } from "react-redux";
import classNames from "classnames/bind";
import { useHistory, useLocation } from "react-router-dom";
import { setSearchValue } from "../../redux/actions/appAction";
import { movieDetails } from "../../types/movie";
import { RootState } from "../../redux/store";

export const FilmCard = (props: { el: movieDetails }) => {
  const dispatch = useDispatch();
  const { genresMap } = useSelector((state: RootState) => state.app);
  const history = useHistory();
  const location = useLocation();
  const openFilm = (id: number) => {
    if (location.pathname.indexOf("search") !== -1) {
      dispatch(setSearchValue(""));
    }
    history.push(`/movie/${id}`);
  };
  const cx = classNames.bind(styles);
  const filmCardClass = cx(
    "rating",
    { rating_up: props.el.vote_average >= 7 },
    { rating_down: props.el.vote_average > 1 && props.el.vote_average < 7 },
    { rating_hide: props.el.vote_average <= 1 }
  );
  return (
    <div className={styles.card} onClick={() => openFilm(props.el.id)}>
      <div className={filmCardClass}>
        {props.el.vote_average && props.el.vote_average.toFixed(1)}
      </div>
      <div className={styles.film_cover}>
        <img
          src={getFilmCover(props.el.poster_path)}
          className={styles.film_cover}
          alt={props.el.title}
          onClick={() => openFilm(props.el.id)}
        />
        <img src={play} className={styles.block} alt="play" />
      </div>
      {props.el.title && <div className={styles.title}>{props.el.title}</div>}
      {props.el.genre_ids && (
        <div className={styles.genres}>
          {genresMap &&
            genresIndexToString(props.el.genre_ids, genresMap).join(" ")}
        </div>
      )}
    </div>
  );
};
