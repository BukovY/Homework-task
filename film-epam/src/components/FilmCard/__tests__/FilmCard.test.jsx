import React from "react";
import { FilmCard } from "../FilmCard";
import { wrapperProvider } from "../../../utils/functrions";
import rerender from "react-test-renderer";

const el = {
  title: "Title film",
  vote_average: 7.8,
};
describe("Testing <FilmCard/> Component", () => {
  it("renders correctly", () => {
    const tree = rerender
      .create(wrapperProvider(<FilmCard el={el} />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
