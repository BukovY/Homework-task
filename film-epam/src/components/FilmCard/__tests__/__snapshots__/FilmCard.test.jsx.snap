// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <FilmCard/> Component renders correctly 1`] = `
<div
  className="card"
  onClick={[Function]}
>
  <div
    className="rating rating_up"
  >
    7.8
  </div>
  <div
    className="film_cover"
  >
    <img
      alt="Title film"
      className="film_cover"
      onClick={[Function]}
      src="https://upload.wikimedia.org/wikipedia/commons/6/65/No-Image-Placeholder.svg"
    />
    <img
      alt="play"
      className="block"
      src="play.svg"
    />
  </div>
  <div
    className="title"
  >
    Title film
  </div>
</div>
`;
